src/interpreter/executors/main.rs

```diff
@@ -30,6 +30,7 @@ impl Interpreter<PredicateStorage> {
             return false;
         }
 
+        #[allow(clippy::needless_collect)] // TODO: the collect could probably be removed
```

Comment:
again; this exists only for code readability and will cause zero runtime overhead. With the current code, its unambiguous that we are extracting memory ranges that are representations of predicate programs.

Reply:
> will cause zero runtime overhead

I'm not sure if this is true, although I'd think the compiler is cabable of omitting the extra allocation. Anyway, I still agree that the current code is more readable.